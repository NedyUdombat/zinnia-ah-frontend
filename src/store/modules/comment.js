import { http } from '../../api/client';

export const LIKE_COMMENT_ERROR = 'LIKE_COMMENT_ERROR';
export const LIKE_COMMENT_SUCCESS = 'LIKE_COMMENT_SUCCESS';

export const likeCommentError = error => ({
  type: LIKE_COMMENT_ERROR,
  error,
});

export const likeCommentSuccess = response => ({
  type: LIKE_COMMENT_SUCCESS,
  response,
});

export const likeCommentRequest = (articleId, commentId) => async dispatch => {
  try {
    const token = localStorage.getItem('token');
    const res = await http.post(
      `/articles/${articleId}/comments/${commentId}/like`,
      {
        headers: {
          Authorization: token,
        },
      },
    );
    if (res.message === 'You have liked this post') {
      return dispatch(likeCommentSuccess(true));
    }
    return dispatch(likeCommentSuccess(false));
  } catch (error) {
    return dispatch(likeCommentError(error));
  }
};

export const DEFAULT_STATE = {
  isLiked: false,
  error: {},
  isLoading: true,
};

export const commentReducer = (state = DEFAULT_STATE, action) => {
  switch (action.type) {
    case LIKE_COMMENT_SUCCESS:
      return {
        ...state,
        isLiked: action.response,
        isLoading: false,
      };
    case LIKE_COMMENT_ERROR:
      return {
        ...state,
        error: action.error,
        isLoading: false,
      };
    default:
      return state;
  }
};
