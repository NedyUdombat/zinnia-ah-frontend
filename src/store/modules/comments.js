import { http } from '../../api/client';
import uniqueBy from 'unique-by';

export const SAVE_COMMENTS = 'SAVE_COMMENTS';
export const CREATE_COMMENT_REQUEST = 'CREATE_COMMENT_REQUEST';
export const CREATE_COMMENT_ERROR = 'CREATE_COMMENT_ERROR';
export const REQUEST_PROCESSING = 'REQUEST_PROCESSING';
export const COMMENT_REACTION_REQUEST = 'COMMENT_REACTION_REQUEST';
export const UNLIKE_COMMENT_REQUEST = 'UNLIKE_REACTION_REQUEST';
export const COMMENT_REACTION_SUCCESS = 'COMMENT_REACTION_SUCCESS';
export const COMMENT_REACTION_ERROR = 'COMMENT_REACTION_ERROR';

export const saveComments = comments => ({
  type: SAVE_COMMENTS,
  comments,
});

export const createCommentError = error => ({
  type: CREATE_COMMENT_ERROR,
  error,
});

export const commentReactionSuccess = (comments, commentId, reaction) => ({
  type: COMMENT_REACTION_SUCCESS,
  comments,
  commentId,
  reaction,
});

export const commentReactionError = error => ({
  type: COMMENT_REACTION_ERROR,
  error,
});

const updateLikes = (comments, commentId, reaction) => {
  const addLike = likes => {
    likes.push({
      userId: getUserId(),
      commentId,
    });
    return likes;
  };

  const removeLike = likes => {
    return likes.filter(like => {
      like.userId !== getUserId();
    });
  };

  comments.forEach(comment => {
    if (comment.id === commentId) {
      comment.likes =
        reaction === 'like'
          ? addLike(comment.likes)
          : removeLike(comment.likes);
    }
  });

  return comments;
};

export const reactToComment = options => async dispatch => {
  const { comments, commentId, articleId } = options;

  try {
    const { data } = await http.post(
      `/articles/${articleId}/comments/${commentId}/like`,
    );
    if (data.message === 'You have liked this post') {
      return dispatch(commentReactionSuccess(comments, commentId, 'like'));
    }
    dispatch(commentReactionSuccess(comments, commentId, 'unlike'));
  } catch (error) {
    dispatch(commentReactionError(error.response));
  }
};

export const requestProcessing = () => ({
  type: REQUEST_PROCESSING,
});

export const createComment = data => async dispatch => {
  dispatch(requestProcessing());
  const { articleId, comment } = data;
  try {
    const {
      data: {
        data: { comment: createdComment },
      },
    } = await http.post(`/articles/${articleId}/comments`, {
      comment,
    });

    createdComment.articleId = articleId;
    createdComment.likes = [];
    return dispatch(saveComments([createdComment]));
  } catch (error) {
    return dispatch(createCommentError(error.response));
  }
};

const getUserId = () => {
  const {
    data: { id: userId },
  } = JSON.parse(localStorage.getItem('userprofile'));
  return userId;
};

export const initialState = {
  isLoading: false,
  userId: getUserId(),
  comments: [
    {
      id: '0c034589-ba07-45fb-b383-9c4e516beb19',
      userId: '0298932c-d9ab-4d32-abca-be92a771ef2c',
      articleId: '141f4f05-7d81-4593-ab54-e256c1006210',
      body: 'My second comment here',
      createdAt: '2019-05-20T19:20:33.191Z',
      updatedAt: '2019-05-20T19:20:33.191Z',
      author: {
        username: 'musonant',
        image: null,
      },
      likes: [
        {
          userId: '0298932c-d9ab-4d31-abca-be92a771ef2c',
          commentId: '0c034589-ba07-45fb-b383-9c4e516beb19',
          createdAt: '2019-05-21T11:46:46.684Z',
          updatedAt: '2019-05-21T11:46:46.684Z',
        },
        {
          userId: '0298932c-d9ab-4d32-abca-be92a771ef2c',
          commentId: '0c034589-ba07-45fb-b383-9c4e516beb19',
          createdAt: '2019-05-21T11:46:46.684Z',
          updatedAt: '2019-05-21T11:46:46.684Z',
        },
      ],
    },
    {
      id: '5d17614e-5a51-47ea-a6eb-63853a4634ee',
      userId: '0298932c-d9ab-4d32-abca-be92a771ef2c',
      articleId: '141f4f05-7d81-4593-ab54-e256c1006210',
      body: 'My first initial comment here',
      createdAt: '2019-05-20T19:25:21.782Z',
      updatedAt: '2019-05-20T19:25:21.782Z',
      author: {
        username: 'musonant',
        image: null,
      },
      likes: [],
    },
  ],
};
export const commentReducer = (state = initialState, action) => {
  switch (action.type) {
    case SAVE_COMMENTS:
      return {
        ...state,
        isLoading: false,
        comments: uniqueBy([...state.comments, ...action.comments], 'id'),
      };
    case REQUEST_PROCESSING:
      return { ...state, isLoading: true };
    case COMMENT_REACTION_SUCCESS:
      return {
        ...state,
        comments: updateLikes(
          action.comments,
          action.commentId,
          action.reaction,
        ),
      };
    default:
      return state;
  }
};
