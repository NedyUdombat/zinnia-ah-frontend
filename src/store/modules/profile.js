import { toast } from 'react-toastify';

import { http } from '../../api/client';
import {
  getProfileRequest,
  deleteArticleRequest,
  getBookmarksRequest,
  getPopularAuthorsRequest,
  postFollowAuthor,
} from '../../api/profile';

export const GET_PROFILE_ERROR = 'GET_PROFILE_ERROR';
export const GET_PROFILE_SUCCESS = 'GET_PROFILE_SUCCESS';
export const DELETE_ARTICLE_PROCESS = 'DELETE_ARTICLE_PROCESS';
export const DELETE_ARTICLE_SUCCESS = 'DELETE_ARTICLE_SUCCESS';
export const DELETE_ARTICLE_ERROR = 'DELETE_ARTICLE_ERROR';
export const UPDATE_USER_PROFILE_ERROR = 'UPDATE_USER_PROFILE_ERROR';
export const UPDATE_USER_PROFILE_SUCCESS = 'UPDATE_USER_PROFILE_SUCCESS';
export const UPDATE_IMAGE_ERROR = 'UPDATE_IMAGE_ERROR';
export const UPDATE_IMAGE_SUCCESS = 'UPDATE_IMAGE_SUCCESS';
export const GET_BOOKMARKS_PROCESS = 'GET_BOOKMARKS_PROCESS';
export const GET_BOOKMARKS_SUCCESS = 'GET_BOOKMARKS_SUCCESS';
export const GET_BOOKMARKS_ERROR = 'GET_BOOKMARKS_ERROR';
export const GET_POPULAR_AUTHORS_SUCCESS = 'GET_POPULAR_AUTHORS_SUCCESS';
export const GET_POPULAR_AUTHORS_ERROR = 'GET_POPULAR_AUTHORS_ERROR';
export const FOLLOW_AUTHOR_PROCESS = 'FOLLOW_AUTHOR_PROCESS';
export const FOLLOW_AUTHOR_SUCCESS = 'FOLLOW_AUTHOR_SUCCESS';
export const FOLLOW_AUTHOR_ERROR = 'FOLLOW_AUTHOR_ERROR';

export const getUserProfileError = error => ({
  type: GET_PROFILE_ERROR,
  error,
});

export const getUserProfile = profile => ({
  type: GET_PROFILE_SUCCESS,
  publications: profile.publications,
  id: profile.id,
  firstName: profile.firstName,
  lastName: profile.lastName,
  bio: profile.bio,
  followings: profile.followings,
  followers: profile.followers,
  image: profile.image,
  email: profile.email,
  username: profile.username,
  updatedAt: profile.updatedAt,
});

export const deleteArticleProcess = () => ({
  type: DELETE_ARTICLE_PROCESS,
});

export const deleteArticleSuccess = publications => ({
  type: DELETE_ARTICLE_SUCCESS,
  publications,
});

export const deleteArticleError = error => ({
  type: DELETE_ARTICLE_ERROR,
  error,
});

export const getBookmarksProcess = () => ({
  type: GET_BOOKMARKS_PROCESS,
});

export const getBookmarksSuccess = publications => ({
  type: GET_BOOKMARKS_SUCCESS,
  publications,
});

export const getBookmarksError = error => ({
  type: GET_BOOKMARKS_ERROR,
  error,
});

export const updateUserProfileError = error => ({
  type: UPDATE_USER_PROFILE_ERROR,
  error,
});

export const updateUserProfile = profile => ({
  type: UPDATE_USER_PROFILE_SUCCESS,
  ...profile,
});

export const getPopularAuthorsSuccess = authors => ({
  type: GET_POPULAR_AUTHORS_SUCCESS,
  authors,
});

export const getPopularAuthorsError = error => ({
  type: GET_POPULAR_AUTHORS_ERROR,
  error,
});

export const followAuthorProcess = () => ({
  type: FOLLOW_AUTHOR_PROCESS,
});

export const followAuthorSuccess = response => ({
  type: FOLLOW_AUTHOR_SUCCESS,
  response: response.data.followers,
});

export const followAuthorError = () => ({
  type: FOLLOW_AUTHOR_ERROR,
});

export const getUserProfileRequest = username => async dispatch => {
  try {
    const res = await getProfileRequest(username);
    dispatch(getUserProfile(res.data.data));
  } catch (error) {
    return dispatch(getUserProfileError(error));
  }
};

export const deleteArticle = (articleId, articles) => async dispatch => {
  try {
    const newArticles = articles.filter(article => article.id !== articleId);
    dispatch(deleteArticleProcess());
    await deleteArticleRequest(articleId);
    dispatch(deleteArticleSuccess(newArticles));
    toast.success('Article has been successfully deleted');
  } catch (error) {
    dispatch(deleteArticleError(error));
    toast.error(error.message);
  }
};

export const updateUserProfileRequest = formData => async dispatch => {
  try {
    const { data } = await http.put('/users/profile/', formData);
    if (data.status === 'success') {
      toast.success('Profile updated successfully');
      return dispatch(updateUserProfile(data.data));
    }
    toast.error('Sorry could not updated profile');
    return dispatch(updateUserProfileError(data.message));
  } catch (error) {
    return error.response;
  }
};

export const getBookmarks = () => async dispatch => {
  try {
    dispatch(getBookmarksProcess());
    const { data } = await getBookmarksRequest();
    dispatch(getBookmarksSuccess(data.data.bookmarks));
  } catch (error) {
    dispatch(getBookmarksError(error));
  }
};

export const getPopularAuthors = () => async dispatch => {
  try {
    const { data } = await getPopularAuthorsRequest();
    dispatch(getPopularAuthorsSuccess(data.data));
  } catch (error) {
    dispatch(getPopularAuthorsError(error));
  }
};

export const followAuthorRequest = (
  followState,
  username,
) => async dispatch => {
  try {
    dispatch(followAuthorProcess());
    const { data } = await postFollowAuthor(followState, username);
    dispatch(followAuthorSuccess(data));
  } catch (error) {
    dispatch(followAuthorError());
    toast.error(`${followState} request was unsuccessful`);
  }
};

export const DEFAULT_STATE = {
  id: '',
  firstName: '',
  lastName: '',
  bio: '',
  publications: [],
  followings: [],
  followers: [],
  bookmarks: [],
  image: '',
  email: '',
  username: '',
  updatedAt: '',
  error: {},
  isLoading: true,
  isButtonLoading: false,
  isDeleting: false,
  isGettingPopularAuthorsLoading: true,
  authors: [],
};

export const profileReducer = (state = DEFAULT_STATE, action) => {
  switch (action.type) {
    case GET_PROFILE_SUCCESS:
    case UPDATE_USER_PROFILE_SUCCESS:
    case UPDATE_IMAGE_SUCCESS:
      return {
        ...state,
        publications: action.publications,
        id: action.id,
        firstName: action.firstName,
        lastName: action.lastName,
        bio: action.bio,
        followings: action.followings,
        followers: action.followers,
        image: action.image,
        email: action.email,
        username: action.username,
        updatedAt: action.updatedAt,
        isLoading: false,
      };
    case GET_PROFILE_ERROR:
    case UPDATE_USER_PROFILE_ERROR:
    case UPDATE_IMAGE_ERROR:
      return {
        ...state,
        error: action.error,
        isLoading: false,
      };
    case DELETE_ARTICLE_PROCESS:
      return {
        ...state,
        isDeleting: true,
      };
    case DELETE_ARTICLE_SUCCESS:
      return {
        ...state,
        isDeleting: false,
        publications: action.publications,
      };
    case DELETE_ARTICLE_ERROR:
      return {
        ...state,
        error: action.error,
        isDeleting: false,
      };
    case GET_BOOKMARKS_PROCESS:
      return {
        ...state,
        isLoading: true,
      };
    case GET_BOOKMARKS_SUCCESS:
      return {
        ...state,
        bookmarks: action.publications,
        isLoading: false,
      };
    case GET_BOOKMARKS_ERROR:
      return {
        ...state,
        error: action.error,
        isLoading: false,
      };
    case GET_POPULAR_AUTHORS_SUCCESS:
      return {
        ...state,
        isGettingPopularAuthorsLoading: false,
        authors: action.authors,
      };
    case GET_POPULAR_AUTHORS_ERROR:
      return {
        ...state,
        isGettingPopularAuthorsLoading: false,
        error: action.error,
      };
    case FOLLOW_AUTHOR_PROCESS:
      return {
        ...state,
        isButtonLoading: true,
      };
    case FOLLOW_AUTHOR_SUCCESS:
      return {
        ...state,
        followers: action.response,
        isButtonLoading: false,
      };
    case FOLLOW_AUTHOR_ERROR:
      return {
        ...state,
        isButtonLoading: false,
      };
    default:
      return state;
  }
};
