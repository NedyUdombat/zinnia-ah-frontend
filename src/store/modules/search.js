import { customSearchRequest } from '../../api/search';

//constants
export const SEARCH_SUCCESS = 'SEARCH_SUCCESS';
export const SEARCH_ERROR = 'SEARCH_ERROR';
export const SEARCH_INITIALIZED = 'SEARCH_INITIALIZED';

export const initialState = {
  isLoading: false,
  errorResponse: [],
  successResponse: {
    data: {
      articles: [],
      authors: [],
      tags: [],
    },
  },
};

export const searchIntialize = () => {
  return {
    type: SEARCH_INITIALIZED,
  };
};

export const searchSuccess = response => {
  return {
    type: SEARCH_SUCCESS,
    response,
  };
};

export const searchError = error => {
  return {
    type: SEARCH_ERROR,
    error,
  };
};

export const customSearch = keyword => {
  return async dispatch => {
    try {
      dispatch(searchIntialize());
      const { data } = await customSearchRequest(keyword);
      dispatch(searchSuccess(data));
    } catch (error) {
      const { data } = error.response;
      dispatch(searchError([data]));
    }
  };
};

export const searchReducer = (state = initialState, action) => {
  switch (action.type) {
    case SEARCH_INITIALIZED:
      return {
        ...state,
        isLoading: true,
      };

    case SEARCH_SUCCESS:
      return {
        ...state,
        successResponse: action.response,
        isLoading: false,
        errorResponse: [],
      };

    case SEARCH_ERROR:
      return {
        ...state,
        isLoading: false,
        errorResponse: action.error,
      };

    default:
      return state;
  }
};
