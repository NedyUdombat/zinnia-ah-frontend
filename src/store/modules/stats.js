import { getReadingStats } from '../../api/users';
import { getToken, decodeToken } from '../../api/helpers';

export const GET_READS_COUNT = 'GET_READS_COUNT';
export const GET_HITS_COUNT = 'GET_HITS_COUNT';
export const GET_STATS_ERROR = 'GET_STATS_ERROR';

export const getUserReadStats = data => {
  return {
    type: GET_READS_COUNT,
    data,
  };
};

export const getStatsError = error => {
  return {
    type: GET_STATS_ERROR,
    error,
  };
};

export const getUserReadingStats = () => {
  return async dispatch => {
    try {
      const {
        data: {
          data: { count, rows },
        },
      } = await getReadingStats();
      const token = getToken();
      const { id } = decodeToken(token);
      // console.log(rows[0].article.userId);
      const hits = rows.filter(row => row.article.userId === id);
      const reads = rows.filter(row => row.article.userId !== id);

      dispatch(
        getUserReadStats({ count, articles: rows, stats: { hits, reads } }),
      );
    } catch (error) {
      const { data } = error.response;
      dispatch(getStatsError([data]));
    }
  };
};

export const initialState = {
  hits: [],
  reads: [],
};

export const statsReducer = (state = initialState, action) => {
  switch (action.type) {
    case GET_READS_COUNT:
      return {
        ...state,
        hits: action.data.stats.hits,
        reads: action.data.stats.reads,
      };
    case GET_STATS_ERROR:
      return {
        ...state,
        error: action.error,
      };
    default:
      return state;
  }
};
