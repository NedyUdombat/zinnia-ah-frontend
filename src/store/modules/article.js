import {
  articleRequest,
  getTrendingArticlesRequest,
  likeArticleRequest,
  fetchArticle,
} from '../../api/article';
import { saveComments } from './comments';

//constants
export const CREATE_ARTCLE_INITIALIZED = 'CREATE_ARTCLE_INITIALIZED';
export const CREATE_ARTICLE_SUCCESS = 'CREATE_ARTICLE_SUCCESS';
export const CREATE_ARTICLE_ERROR = 'CREATE_ARTICLE_ERROR';
export const LIKE_ARTCLE_INITIALIZED = 'LIKE_ARTCLE_INITIALIZED';
export const LIKE_ARTICLE_SUCCESS = 'LIKE_ARTICLE_SUCCESS';
export const LIKE_ARTICLE_ERROR = 'LIKE_ARTICLE_ERROR';
export const GET_TRENDING_ARTICLES_SUCCESS = 'GET_TRENDING_ARTICLES_SUCCESS';
export const GET_TRENDING_ARTICLES_ERROR = 'GET_TRENDING_ARTICLES_ERROR';

export const FETCH_ARTICLE_START = 'FETCH_ARTICLE_START';
export const FETCH_ARTICLE_SUCCESS = 'FETCH_ARTICLE_SUCCESS';
export const FETCH_ARTICLE_FAILURE = 'FETCH_ARTICLE_FAILURE';

export const createArticleInitialized = () => {
  return {
    type: CREATE_ARTCLE_INITIALIZED,
  };
};

export const createArticleSuccess = response => {
  return {
    type: CREATE_ARTICLE_SUCCESS,
    response,
  };
};

export const createArticleError = error => {
  return {
    type: CREATE_ARTICLE_ERROR,
    error,
  };
};

export const getArticleStart = () => {
  return {
    type: FETCH_ARTICLE_START,
  };
};

export const getArticleSuccess = article => {
  return {
    type: FETCH_ARTICLE_SUCCESS,
    article,
  };
};

export const getArticleFailure = error => {
  return {
    type: FETCH_ARTICLE_FAILURE,
    error,
  };
};

export const likeArticleInitialized = () => {
  return {
    type: LIKE_ARTCLE_INITIALIZED,
  };
};

export const likeArticleSuccess = response => {
  return {
    type: LIKE_ARTICLE_SUCCESS,
    response,
  };
};

export const likeArticleError = error => {
  return {
    type: LIKE_ARTICLE_ERROR,
    error,
  };
};

export const getTrendingArticlesSuccess = trendingArticles => ({
  type: GET_TRENDING_ARTICLES_SUCCESS,
  trendingArticles,
});

export const getTrendingArticlesError = error => ({
  type: GET_TRENDING_ARTICLES_ERROR,
  error,
});

export const initialState = {
  isLoading: false,
  errorResponse: [],
  successResponse: {
    status: '',
  },
  trendingArticles: [],
  isGettingTrendingArticles: true,
  articles: {},
};

export const createArticle = articleData => {
  return async dispatch => {
    try {
      dispatch(createArticleInitialized());
      const { data } = await articleRequest(articleData);
      location.href = `/read/${data.data.slug}`;
      dispatch(createArticleSuccess(data));
    } catch (error) {
      const { data } = error.response;
      dispatch(createArticleError([data]));
    }
  };
};

export const getSingleArticle = (articleId, history) => {
  return async dispatch => {
    dispatch(getArticleStart());
    try {
      const { data } = await fetchArticle(articleId);
      dispatch(getArticleSuccess(data.data));
      dispatch(saveComments(data.data.comments));
    } catch ({ response }) {
      response.status === 404 ? history.push('/404') : '';
      dispatch(getArticleFailure(response.data));
    }
  };
};

export const likeArticle = action => {
  return async dispatch => {
    try {
      dispatch(likeArticleInitialized());
      const { data } = await likeArticleRequest(action);
      dispatch(likeArticleSuccess(data));
    } catch (error) {
      const { data } = error.response;
      dispatch(likeArticleError([data]));
    }
  };
};

export const getTrendingArticles = () => async dispatch => {
  try {
    const { data } = await getTrendingArticlesRequest();
    dispatch(getTrendingArticlesSuccess(data.data.rows));
  } catch (error) {
    dispatch(getTrendingArticlesError(error.response));
  }
};

export const articleReducer = (state = initialState, action) => {
  switch (action.type) {
    case FETCH_ARTICLE_START:
      return {
        ...state,
        isLoading: true,
      };

    case CREATE_ARTICLE_SUCCESS:
      return {
        ...state,
        successResponse: action.response,
        isLoading: false,
        errorResponse: [],
      };

    case CREATE_ARTCLE_INITIALIZED:
      return {
        ...state,
        isLoading: true,
      };

    case CREATE_ARTICLE_ERROR:
      return {
        ...state,
        isLoading: false,
        errorResponse: action.error,
      };

    case FETCH_ARTICLE_SUCCESS:
      return {
        ...state,
        isLoading: false,
        articles: {
          ...state.articles,
          [action.article.slug]: action.article,
        },
      };

    case FETCH_ARTICLE_FAILURE:
      return {
        ...state,
        isLoading: false,
        error: action.error,
      };
    case LIKE_ARTCLE_INITIALIZED:
      return {
        ...state,
        isLoading: true,
      };

    case LIKE_ARTICLE_SUCCESS:
      return {
        ...state,
        successResponse: action.response,
        isLoading: false,
        errorResponse: [],
      };

    case LIKE_ARTICLE_ERROR:
      return {
        ...state,
        isLoading: false,
        errorResponse: action.error,
      };

    case GET_TRENDING_ARTICLES_SUCCESS:
      return {
        ...state,
        isGettingTrendingArticles: false,
        trendingArticles: action.trendingArticles,
      };
    case GET_TRENDING_ARTICLES_ERROR:
      return {
        ...state,
        isGettingTrendingArticles: false,
        error: action.error,
      };
    default:
      return state;
  }
};
