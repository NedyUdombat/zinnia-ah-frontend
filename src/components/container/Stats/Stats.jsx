import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { Tab, Label, Menu } from 'semantic-ui-react';

import { getUserReadingStats } from '../../../store/modules/stats';

// styles
import './Stats.scss';

//components
import ReadingList from '../../presentationals/ReadingLists/ReadingList';

const Stats = props => {
  const {
    stats: { hits, reads },
    getUserReadStats,
  } = props;

  const getStats = async () => {
    await getUserReadStats();
  };

  React.useEffect(() => {
    getStats();
  }, []);

  const panes = [
    {
      menuItem: (
        <Menu.Item key="reads">
          READS<Label>{reads.length}</Label>
        </Menu.Item>
      ),
      render: () => (
        <Tab.Pane>
          <div
            style={{
              marginTop: '0.0625rem',
            }}
          >
            <ReadingList articles={reads} type="reads" />
          </div>
        </Tab.Pane>
      ),
    },
    {
      menuItem: (
        <Menu.Item key="hits">
          HITS<Label>{hits.length}</Label>
        </Menu.Item>
      ),
      render: () => (
        <Tab.Pane>
          <div
            style={{
              marginTop: '0.0625rem',
            }}
          >
            <ReadingList articles={hits} type="hits" />
          </div>
        </Tab.Pane>
      ),
    },
  ];

  return (
    <div className="count">
      <div className="stats-tab-menu">
        <Tab menu={{ secondary: true, pointing: true }} panes={panes} />
      </div>
    </div>
  );
};
const mapStateToProps = state => ({
  stats: state.stats,
});
const mapDispatchToProps = dispatch => ({
  getUserReadStats: () => dispatch(getUserReadingStats()),
});
Stats.propTypes = {
  stats: PropTypes.object.isRequired,
  getUserReadStats: PropTypes.func,
};

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(Stats);
