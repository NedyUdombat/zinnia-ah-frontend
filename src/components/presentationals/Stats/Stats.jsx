import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { Tab, Label, Menu } from 'semantic-ui-react';
// import ArticleLists from '../ArticleLists/ArticleLists';
// import Title from '../Title/Title';
import {
  getUserReadingStats,
  statsReducer,
} from '../../../store/modules/stats';

// styles
import './Stats.scss';

const Stats = props => {
  const { readStats, getUserReadStats } = props;
  const getStats = async () => {
    await getUserReadStats();
  };
  React.useEffect(() => {
    getStats();
  }, []);
  const panes = [
    {
      menuItem: (
        <Menu.Item key="reads">
          READS<Label>8{readStats}</Label>
        </Menu.Item>
      ),
      render: () => (
        <Tab.Pane>
          <div
            style={{
              marginTop: '100px',
            }}
          >
            Reads
            {/* <ReadingList /> */}
          </div>
        </Tab.Pane>
      ),
    },
    {
      menuItem: (
        <Menu.Item key="hits">
          HITS<Label>20</Label>
        </Menu.Item>
      ),
      render: () => <Tab.Pane>Hits</Tab.Pane>,
    },
  ];
  return (
    <div className="count">
      <div className="stats-tab-menu">
        <Tab menu={{ secondary: true, pointing: true }} panes={panes} />
      </div>
    </div>
  );
};
const mapStateToProps = () => ({
  articlesRead: statsReducer.response,
});
const mapDispatchToProps = dispatch => ({
  getUserReadStats: () => dispatch(getUserReadingStats()),
});
Stats.propTypes = {
  readStats: PropTypes.number,
  getUserReadStats: PropTypes.func,
};

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(Stats);
