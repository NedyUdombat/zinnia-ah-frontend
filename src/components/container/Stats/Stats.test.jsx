import React from 'react';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';

import Stats from '../Stats/Stats';
import { initialState } from '../../../store/modules/stats';

const props = {
  stats: { hits: [], reads: [] },
  getUserReadStats: jest.fn(),
};

const mockStore = configureStore([thunk]);
const store = mockStore(initialState);
describe('Stats test suite', () => {
  it('should render correctly', () => {
    const component = mount(
      <Provider store={store}>
        <Stats {...props} />
      </Provider>,
    );
    expect(component).toMatchSnapshot();
  });
});
