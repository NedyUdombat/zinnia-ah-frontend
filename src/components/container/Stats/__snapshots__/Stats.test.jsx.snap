// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stats test suite should render correctly 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <ConnectFunction
    getUserReadStats={[MockFunction]}
    stats={
      Object {
        "hits": Array [],
        "reads": Array [],
      }
    }
  >
    <Stats
      dispatch={[Function]}
      getUserReadStats={[Function]}
      stats={
        Object {
          "hits": Array [],
          "reads": Array [],
        }
      }
    >
      <div
        className="count"
      >
        <div
          className="stats-tab-menu"
        >
          <Tab
            grid={
              Object {
                "paneWidth": 12,
                "tabWidth": 4,
              }
            }
            menu={
              Object {
                "pointing": true,
                "secondary": true,
              }
            }
            panes={
              Array [
                Object {
                  "menuItem": <MenuItem>
                    READS
                    <Label>
                      0
                    </Label>
                  </MenuItem>,
                  "render": [Function],
                },
                Object {
                  "menuItem": <MenuItem>
                    HITS
                    <Label>
                      0
                    </Label>
                  </MenuItem>,
                  "render": [Function],
                },
              ]
            }
            renderActiveOnly={true}
          >
            <div>
              <Menu
                activeIndex={0}
                items={
                  Array [
                    <MenuItem>
                      READS
                      <Label>
                        0
                      </Label>
                    </MenuItem>,
                    <MenuItem>
                      HITS
                      <Label>
                        0
                      </Label>
                    </MenuItem>,
                  ]
                }
                onItemClick={[Function]}
                pointing={true}
                secondary={true}
              >
                <div
                  className="ui pointing secondary menu"
                >
                  <MenuItem
                    active={true}
                    index={0}
                    key="reads"
                    onClick={[Function]}
                  >
                    <a
                      className="active item"
                      onClick={[Function]}
                    >
                      READS
                      <Label>
                        <div
                          className="ui label"
                          onClick={[Function]}
                        >
                          0
                        </div>
                      </Label>
                    </a>
                  </MenuItem>
                  <MenuItem
                    active={false}
                    index={1}
                    key="hits"
                    onClick={[Function]}
                  >
                    <a
                      className="item"
                      onClick={[Function]}
                    >
                      HITS
                      <Label>
                        <div
                          className="ui label"
                          onClick={[Function]}
                        >
                          0
                        </div>
                      </Label>
                    </a>
                  </MenuItem>
                </div>
              </Menu>
              <TabPane
                active={true}
                as={[Function]}
              >
                <Segment
                  attached="bottom"
                  className="active tab"
                >
                  <div
                    className="ui bottom attached segment active tab"
                  >
                    <div
                      style={
                        Object {
                          "marginTop": "0.0625rem",
                        }
                      }
                    >
                      <ReadingList
                        articles={Array []}
                        type="reads"
                      >
                        <div
                          className="post-lists"
                        >
                          <p
                            style={
                              Object {
                                "fontSize": "30px",
                                "fontWeight": "bolder",
                              }
                            }
                          >
                            You have no reads
                          </p>
                        </div>
                      </ReadingList>
                    </div>
                  </div>
                </Segment>
              </TabPane>
            </div>
          </Tab>
        </div>
      </div>
    </Stats>
  </ConnectFunction>
</Provider>
`;
